File: problems/P-35-search-insert-position/solution.py
"""
LeetCode Problem: 35. Search Insert Position
Difficulty: Easy
Topics: Array, Binary Search
Date Solved: 30/05/2025
"""

class Solution:
    def searchInsert(self, nums: list[int], target: int) -> int:
        for i in range(len(nums)):
            if nums[i] >= target:
                return i
        
        return len(nums)

File: problems/P-35-search-insert-position/solution.cpp
/*
 * LeetCode Problem: 35. Search Insert Position
 * Difficulty: Easy
 * Topics: Array, Binary Search
 * Date Solved: 30/05/2025
*/

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if (nums.size() == 0) return 0;

        int low = 0, high = nums.size() - 1;
        while (low <= high) {
            int mid = low + (high - low) / 2;

            if (target == nums[mid])
                return mid;
            else if (target > nums[mid])
                low = mid + 1;
            else
                high = mid - 1;
        }

        return low;
    }
};
